src/js/models/layerstructure.js

```diff
@@ -1201,36 +1201,57 @@ define(function (require, exports, module) {
      */
     LayerStructure.prototype.createGroup = function (documentID, groupID, groupEndID, groupName,
         isArtboard, boundsDescriptor) {
+        // Creates the group head and end, needed for any type of group 
         var groupHead = Layer.fromGroupDescriptor(documentID, groupID, groupName, false,
                 isArtboard, boundsDescriptor),
             groupEnd = Layer.fromGroupDescriptor(documentID, groupEndID, "", true),
-            layersToMove = this.selectedNormalized.flatMap(this.descendants, this).toOrderedSet(),
-            layersToMoveIndices = layersToMove.map(this.indexOf, this),
-            layersToMoveIDs = collection.pluck(layersToMove, "id"),
-            groupHeadIndex = this.layers.size - layersToMoveIndices.last(),
-            newGroupIDs = Immutable.Seq([groupEndID, layersToMoveIDs, groupID]).flatten().reverse(),
-            removedIDs = collection
-                .difference(this.index, layersToMoveIDs) // Remove layers being moved
-                .reverse(), // Reverse because we want to slice from the end
-            newIDs = removedIDs
-                .slice(0, groupHeadIndex) // First chunk is all layers up to top most selected one
-                .concat(newGroupIDs) // Then our new group
-                .concat(removedIDs.slice(groupHeadIndex)), // Then the rest
+            groupHeadIndex,
+            updatedLayers,
+            newLayerStructure;
+        
+        if (isArtboard) {
```

Comment:
There is a lot of duplicated code across these two branches. Is it possible to have, e.g., two if statements so that we don't have any duplicated code?